src/systems/systems/src/shared/mod.rs

```diff
@@ -43,11 +43,31 @@ pub trait RamAirTurbineHydraulicLoopPressurised {
     fn is_rat_hydraulic_loop_pressurised(&self) -> bool;
 }
 
-pub trait LandingGearPosition {
+pub trait LandingGearRealPosition {
     fn is_up_and_locked(&self) -> bool;
     fn is_down_and_locked(&self) -> bool;
 }
 
+pub trait LgciuWeightOnWheels {
+    fn right_gear_compressed(&self, treat_ext_pwr_as_ground: bool) -> bool;
+    fn right_gear_extended(&self, treat_ext_pwr_as_ground: bool) -> bool;
+
+    fn left_gear_compressed(&self, treat_ext_pwr_as_ground: bool) -> bool;
+    fn left_gear_extended(&self, treat_ext_pwr_as_ground: bool) -> bool;
+
+    fn left_and_right_gear_compressed(&self, treat_ext_pwr_as_ground: bool) -> bool;
+    fn left_and_right_gear_extended(&self, treat_ext_pwr_as_ground: bool) -> bool;
+
+    fn nose_gear_compressed(&self, treat_ext_pwr_as_ground: bool) -> bool;
+    fn nose_gear_extended(&self, treat_ext_pwr_as_ground: bool) -> bool;
```

Comment:
A note for other reviewers: If you think of these functions as discrete connections going from the specific LGCIU to another system, then in reality there is various ways to encode this. A signal might be truthy when 28V DCC is applied, it might be truthy when GND is applied, and some connections have a special transistor to explicitly treat external power as ground.

While we don't explicitly model the outgoing voltage here (we describe the outgoing logical meaning of a signal), every consumer can individually decide whether it wants the logically positive or negative case (compressed vs. extended), and whether it wants the LGCIU to treat it's specific connection with the special external power ground case. The positive/negative cases are important for system to decide which way to fail in a fault case (default to extended vs. compressed, or flight vs. ground), and each consumer can also decide individually whether external power is good enough to force the ground case.

This is why all these combinations are supplied, and why the external ground power override is modeled as a parameter interpreted inside the LGCIU (rather than the target system).

Reply:
We'd probably want to make in the future a list of the system users and what signal they should use. And mark this as "reality checked" or "Rule of thumb only, awaiting real data confirmation"